### PR TITLE
Validate sort order in admin.ResourceListRequest

### PR DESCRIPTION
**Pull request by iaroslav-ciupin**
# TL;DR
Validate sort order by checking it against supported columns. This PR only checks sort order in `admin.ResourceListRequest`, other requests will be added in separate PR. Also validation for filters will be added in separate PR.

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
_Remove the '*fixes*' keyword if there will be multiple PRs to fix the linked issue_

fixes https://github.com/flyteorg/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/flyteorg/flyte/issues/<number>_

### CODE REVIEW

File: pkg/manager/impl/execution_manager.go [hunk 1]
```diff
@@ -1434,12 +1427,10 @@ func (m *ExecutionManager) ListExecutions(
...
 		return nil, err
 	}
-	var sortParameter common.SortParameter
-	if request.SortBy != nil {
-		sortParameter, err = common.NewSortParameter(*request.SortBy)
-		if err != nil {
-			return nil, err
-		}
+
+	sortParameters, err := common.NewSortParameters(&request, gormimpl.ExecutionColumns)
```
Comment (katrogan) [inline pkg/manager/impl/execution_manager.go:1431]: Because we [join](https://github.com/flyteorg/flyteadmin/blob/master/pkg/repositories/gormimpl/execution_repo.go#L79-L97) on multiple tables for the list request we should support sort columns from any of the supported tables

See https://docs.flyte.org/en/latest/concepts/admin.html#putting-it-all-together for how we document table_name.field as an option sort

Comment (iaroslav-ciupin) [reply pkg/manager/impl/execution_manager.go:1431]: @katrogan from what I see from code and docs, we are joining with tables references in filters, not in sort_by. 
I was able to successfully perform this query `https://dogfood.cloud-staging.union.ai/api/v1/executions/flytesnacks/development?limit=5&filters=eq(workflow.name,ray_example.ray_example.ray_workflow)`

But I got an error for
`https://dogfood.cloud-staging.union.ai/api/v1/executions/flytesnacks/development?limit=5&sort_by.key=workflow.name&filters=eq(workflow.name,ray_example.ray_example.ray_workflow)`
and 
`https://dogfood.cloud-staging.union.ai/api/v1/executions/flytesnacks/development?limit=5&sort_by.key=workflow.name`
Error
```
{
  "error": "cannot query with specified table attributes: missing FROM-clause entry for table \"workflow\"",
  "code": 3,
  "message": "cannot query with specified table attributes: missing FROM-clause entry for table \"workflow\""
}
```

Comment (katrogan) [reply pkg/manager/impl/execution_manager.go:1431]: thanks for the verification @iaroslav-ciupin 
do you mind filing a follow-up issue to ensure that we update [JoinTableEntities](https://github.com/flyteorg/flyteadmin/blob/6c97bb9c5558af240b2d67290bb35393990678cb/pkg/repositories/interfaces/common.go#L27) to include other tables when they're included in the sort key as well?

**Review by katrogan — APPROVED** [review]

File: pkg/manager/impl/execution_manager.go (hunk 1 above)
Comment (katrogan) [reply pkg/manager/impl/execution_manager.go:1431]: Actually, can you still ensure that we pass in the join table columns here as allowed columns?

**Review by katrogan — CHANGES_REQUESTED** [review]

File: pkg/manager/impl/execution_manager.go (hunk 1 above)
Comment (iaroslav-ciupin) [reply pkg/manager/impl/execution_manager.go:1431]: @katrogan I don't get, why? we currently don't support sorting by join table columns and return a DB error for that? wouldn't it be better to return a clearer validation error ?

Comment (katrogan) [reply pkg/manager/impl/execution_manager.go:1431]: ah I see - missed your second example, so even for the successful join call, e.g.
`/api/v1/executions/flytesnacks/development?limit=5&filters=eq(workflow.name,ray_example.ray_example.ray_workflow)&sort_by.key=workflow.created_at&sort_by.direction=DESCENDING`

we still fail to sort by so we're not removing functionality in this change

Comment (iaroslav-ciupin) [reply pkg/manager/impl/execution_manager.go:1431]: @katrogan exactly, although we specify join table column in filter and we still can't sort by join table column
